script.js

```diff
@@ -8,160 +48,114 @@ function shuffleArray(array) {
-                    button2.remove()
-                    button3.remove()
-                    hintButton.remove()
-                    clearInterval(interval)
-                    if (films.length > 2) {
-                        return playGame(filmTitles, films)
-                    } else {
-                        gameOverModal.showModal()
-                    }
-                }, 200)
-            })
-    }
-
-
-    hintContainer.appendChild(hintButton)
-
-    checkAnswer(button1)
-    checkAnswer(button2)
-    checkAnswer(button3)
-
-    // hint
     hintButton.addEventListener('click', () => {
         hintButton.textContent = currentQuote.year
-        hintButton.disabled = true
     })
-    return films
 }
 
-const instructionsButton = document.querySelector('.instructionsButton')
-const instructions = document.querySelector('.modalContainer')
-const closeButton = document.querySelector('.closeButton')
-const modal = () => {
-    if (instructions.style.display === 'none') {
-        instructions.style.display = 'flex';
-        instructions.style.zIndex = '1';
-    }
-    else {
-        instructions.style.display = 'none';
-    }
+const endGame = () => {
+    clearInterval(countDownInterval)
+    scoreSpan.textContent = score.toString()
+    gameOverModal.showModal()
+    countDownValue = 30
+    timerDisplay.textContent = countDownValue.toString()
 }
 
-instructionsButton.addEventListener('click', modal)
-closeButton.addEventListener('click', modal)
-
-const timerDisplay = document.querySelector(".timerDisplay")
-const timerProgress = document.querySelector(".timerProgress")
-const resetBtn = document.querySelector('.reset')
-const gameOverModal = document.querySelector('.gameOverModal')
-const scoreSpan = document.querySelector('.score')
-
-let countDownValue = 30
-let countDownInterval
-
-const countDownTimer = ()=>{countDownInterval=setInterval(countDown,1000)}
+const countDownTimer = () => {
+    countDownInterval = setInterval(countDown, 1000)
+}
 
 const countDown = () => {
-    if(countDownValue > 0){
+    if (countDownValue > 0) {
         countDownValue--
         timerDisplay.textContent = countDownValue.toString()
         timerProgress.value = countDownValue
-    }
-    else {
-        clearInterval(countDownInterval)
-        scoreSpan.textContent = score.toString()
-        gameOverModal.showModal()
-        countDownValue = 30
-        timerDisplay.textContent = countDownValue.toString()
+    } else {
+        endGame()
     }
 }
 
-playButton.addEventListener('click',countDownTimer)
-
-timerDisplay.textContent = countDownValue.toString()
-
 const resetGame = () => {
     score = 0
     countDownValue = 30
     scoreBoxElem.textContent = score
     playGame()
 }
 
-resetBtn.addEventListener('click',resetGame)
-resetBtn.addEventListener('click',countDownTimer)
\ No newline at end of file
+const playGame = () => {
+    setupGame().
+    then(() => roundGenerator(filmTitles,films))
+    countDownTimer()
+}
```

Comment:
So `setupGame` is going to go away and fetch the films, and then start the game eventually, after it has done that.
Once `setupGame` has started, execution is going to continue, and run the `countDownTimer` immediately.
So if our user has a slow connection, is that gong to be fair?
How do you ensure that the countdown only starts at the same time as the first round of the game?